proxy/src/coco/source.rs

```diff
@@ -260,18 +281,66 @@ pub fn local_state(repo_path: &str) -> Result<LocalState, error::Error> {
     Ok(LocalState { branches, managed })
 }
 
-/// Retrieves the [`Commit`] for the given `sha1`.
+/// Retrieves the [`CommitHeader`] for the given `sha1`.
 ///
 /// # Errors
 ///
 /// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
-pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit, error::Error> {
+pub fn commit_header<'repo>(
+    browser: &mut Browser<'repo>,
+    sha1: &str,
+) -> Result<CommitHeader, error::Error> {
     browser.commit(surf::vcs::git::Oid::from_str(sha1)?)?;
 
     let history = browser.get();
     let commit = history.first();
 
-    Ok(Commit::from(commit))
+    Ok(CommitHeader::from(commit))
+}
+
+/// Retrieves a [`Commit`].
+///
+/// # Errors
+///
+/// Will return [`error::Error`] if the project doesn't exist or the surf interaction fails.
+pub fn commit<'repo>(browser: &mut Browser<'repo>, sha1: &str) -> Result<Commit, error::Error> {
```

Comment:
We could have a general case between any two commits as well right? Where this is the specialised case of comparing to the parent?

Reply:
Yeah, though we won't be needing it until we introduce revisions, down the line..

Reply:
Well how about we make a private function that takes two `Oid`s and express `commit` in terms of that?

Reply:
So `pub fn diff<'repo>(browser: &mut Browser<'repo>, left: Option<&str>, right: &str) -> Result<Diff, error::Error>`?

Reply:
I'd be more inclined to do `Oid` instead of `&str` because when you get the `Commit`'s parent it will be an `Oid` instead of `&str` and then you'll end up doing `to_string()` and inside there `from_str`. Goad I hate strings.

Reply:
It's a bit ugly, it's also not clear where the stats code goes then.

Reply:
Ah, I think I see what you mean. Nevermind, we can think about it again :)